### ISSUE TITLE
Yahoo Finance stops working

### ISSUE THREAD

**Issue by seg-on** [iprak/yahoofinance#136]
Hi,
It seems that after updating HA to 2024.10.4, Yahoo Finance stops working.
I've tried rebooting HA, rebooting PC but it doesn't help.

**Comment by iprak** [comment]
I need more info.

You can also enable debug level logging to capture more data

```
  logs:
    custom_components.yahoofinance: debug
```

**Comment by seg-on** [comment]
Hi,
you mean this?


```
2024-10-29 10:20:23.636 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:20:23.637 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:20:53.640 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:20:53.641 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:21:23.645 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:21:23.645 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:21:53.649 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:21:53.649 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:22:23.654 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:22:23.654 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:22:53.659 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:22:53.660 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:23:23.662 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:23:23.663 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:23:53.665 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:23:53.665 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:24:23.669 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting crumb. 
2024-10-29 10:24:23.669 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-10-29 10:24:38.671 DEBUG (MainThread) [custom_components.yahoofinance.coordinator] Navigating to base Yahoo page
```

**Comment by iprak** [comment]
Yes, this has happened when the Yahoo website returns different data. Usually due to GDPR consent. What is your location and the url of the Yahoo website (if opened in a browser) ?

**Comment by seg-on** [comment]
Yes, I'm from Slovakia (part of EU). I don't understand what you mean by Yahoo website URL? When I type finance.yahoo.com I get https://finance.yahoo.com/.

**Comment by seg-on** [comment]
It is strange, but now at 18:11 it has started to work.
I'll be watching that.

**Comment by vellanix** [comment]
Also here (Italy) doesnt work from yesterday afternoon.

Same error: Unable to get crumb

**Comment by rummepa** [comment]
same thing here in Belgium:

Logger: custom_components.yahoofinance
Source: custom_components/yahoofinance/__init__.py:241
integration: Yahoo Finance ([documentation](https://github.com/iprak/yahoofinance), [issues](https://github.com/iprak/yahoofinance/issues))
First occurred: October 29, 2024 at 21:03:19 (3155 occurrences)
Last logged: 10:17:09
Unable to get crumb, re-trying in 15 seconds

**Comment by jan-jeroen** [comment]
Same here in the Netherlands sinterklaas last weekend. Unable to get crumb.

**Comment by iprak** [comment]
The release 1.2.11 should help.

**Comment by jan-jeroen** [comment]
> The release 1.2.11 should help.

Confirmed! Many thanks

**Comment by seg-on** [comment]
I'm sad. I updated to this version and after that it stopped working :(

**Comment by iprak** [comment]
Can you share debug log?

**Comment by seg-on** [comment]
here you are:
```
2024-10-31 20:39:45.002 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:40:15.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:40:15.002 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:40:45.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:40:45.003 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:41:15.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:41:15.002 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:41:45.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:41:45.002 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:42:15.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:42:15.003 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:42:45.002 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B
2024-10-31 20:42:45.003 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
2024-10-31 20:42:47.202 ERROR (MainThread) [custom_components.weatherdotcom.coordinator] Error fetching Weather.com data: TimeoutError()
2024-10-31 20:43:05.001 DEBUG (MainThread) [custom_components.yahoofinance.coordinator] Requesting data from 'https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=FKbLQWwD20B'
```

**Comment by iprak** [comment]
@seg-on  I think you have some other problem. The log indicates similar failure for `weatherdotcom.coordinator` data fetch.

Everything seems to check out from my end when using VPN to fake European source.

**Comment by jan-jeroen** [comment]
My problem is back since yesterday, the new version worked for about a day.

2024-11-01 14:04:18.872 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-11-01 14:04:33.966 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-11-01 14:04:49.072 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds

**Comment by iprak** [comment]
I am not hopeful of addressing this issue. I am not able to replicate this over VPN. But I have pushed out another release with more logging.

**Comment by jan-jeroen** [comment]
Installed your new release. This is what I get:

2024-11-02 10:01:41.962 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Crumb request responded with status=404
2024-11-02 10:01:41.963 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds
2024-11-02 10:01:57.137 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Crumb request responded with status=404
2024-11-02 10:01:57.138 WARNING (MainThread) [custom_components.yahoofinance] Unable to get crumb, re-trying in 15 seconds

**Comment by seg-on** [comment]
Yahoo Finance ver: 1.2.12

```
2024-11-02 13:03:30.003 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Timed out getting data from https://query1.finance.yahoo.com/v7/finance/quote?symbols=EURUSD=X,GC=F,ES=F,MNQ=F,^VIX,CL=F,TTF=F,^GSPC,^NDX,SXR8.DE,BTC-USD,BTC-EUR,XMR-EUR&crumb=N1QwNF/tiD5
2024-11-02 13:03:30.004 WARNING (MainThread) [custom_components.yahoofinance.coordinator] Error obtaining data, retrying in 20 seconds
```

**Comment by iprak** [comment]
I have attempted one more fix. It has been trouble some to even replicate this issue which could be due to how my VPN connection is directing traffic or perhaps I am getting the same remote address.

Sorry but I won't be able to look into this.  One can however fork the project and implement/support this elsewhere.

It was not my intention for the integration to be completely usable outside US. It would be difficult to continuously bypass GDPS restrictions to account for changes. This will also definitely make it not acceptable as a core integration in HomeAssistant which is what I eventually want it to be.

**Comment by seg-on** [comment]
Hi @iprak,
I found out that API v6 is working for me (I am using a AppScript in Google Sheets that use this API to get stock data).
Can you look if there is the big difference between V7 you are using and V6? If it is not possible to port to V6?
https://query2.finance.yahoo.com/v6/finance/options/<ticker>
https://query2.finance.yahoo.com/v6/finance/options/aapl

**Comment by iprak** [comment]
The Yahoo end points are undocumented, this integration is based on my trial an error. v7 url is what is used when accessing finance.yahoo.com from US location, so I cannot tell how is v6 different. One would think that v7 is newer.

Something like this happened in May 2023 (#96) where v6 version was successful temporarily. You can locally change to use v6 but I won't be able to make that part of the integration.

**Comment by applegeeks** [comment]
> This has stopped working this mornin.
> 
> 2024-11-19 10:03:30.129 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Received status 401 (Unauthorized User is unable to access this feature - https://bit.ly/yahoo-finance-api-feedback
> 
> Says API has been disabled
> 
> website gives the following message
> 
> {"finance":{"result":null,"error":{"code":"Unauthorized","description":"User is unable to access this feature - [https://bit.ly/yahoo-finance-api-feedback"}}}](https://bit.ly/yahoo-finance-api-feedback%22%7D%7D%7D)

seems to be up and running again.

**Comment by nexusis7** [comment]
yes can confirm it’s working again

**Comment by iprak** [comment]
This seems to be more of a Yahoo end point issue. What country is this?

**Comment by nexusis7** [comment]
I’m in Canada.

**Comment by iprak** [comment]
I will give it a test using VPN.

**Comment by applegeeks** [comment]
> I’m in Canada.

Australia here.

**Comment by vPotusFlotus** [comment]
> This has stopped working this mornin.
> 
> 2024-11-19 10:03:30.129 ERROR (MainThread) [custom_components.yahoofinance.coordinator] Received status 401 (Unauthorized User is unable to access this feature - https://bit.ly/yahoo-finance-api-feedback
> 
> Says API has been disabled
> 
> website gives the following message
> 
> {"finance":{"result":null,"error":{"code":"Unauthorized","description":"User is unable to access this feature - [https://bit.ly/yahoo-finance-api-feedback"}}}](https://bit.ly/yahoo-finance-api-feedback%22%7D%7D%7D)

I receive the same error as stated above (from Belgium). Seems that API Access to Yahoo Finance might be disabled?
![image](https://github.com/user-attachments/assets/881be8bb-f61b-4c74-88e3-57ffd890d3d6)

At least that's what the URL states from the Error Message: https://bit.ly/yahoo-finance-api-feedback

**Comment by spacelama** [comment]
Might be an idea to change the wording on the readme away from "This integration will most only work in US mainland. GDPR requirements seems to be causing requests to fail".  If it is a GDPR problem, it's more likely it'll work everywhere but Europe, as opposed to only working in the US.

It is currently working in Australia.  But I briefly rolled back to 1.2.10 when it *appeared* it wasn't working (actually a config I broke at the same time), then noticed the wording indicating a change and problem in 1.2.12, thus indicating to me that I'd have more luck with the version I had from my backups.

**Comment by iprak** [comment]
> Might be an idea to change the wording on the readme away from "This integration will most only work in US mainland. GDPR requirements seems to be causing requests to fail". If it is a GDPR problem, it's more likely it'll work everywhere but Europe, as opposed to only working in the US.
> 
> It is currently working in Australia. But I briefly rolled back to 1.2.10 when it _appeared_ it wasn't working (actually a config I broke at the same time), then noticed the wording indicating a change and problem in 1.2.12, thus indicating to me that I'd have more luck with the version I had from my backups.

I have slightly reworded that line.
